compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -283,17 +283,28 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
           val ctx = comparerContext
           given Context = ctx // optimization for performance
           val info2 = tp2.info
+
+          /** Does `tp2` have a singleton type or NoPrefix as a prefix?
+           *  If that's not the case, following an alias via asSeenFrom could be lossy
+           *  so we should not conclude `false` if comparing aliases fails.
+           *  See pos/i17064.scala for a test case
+           */
+          def hasPrecisePrefix(tp: NamedType) =
+            tp.prefix.isSingleton || tp.prefix == NoPrefix
```

Comment:
This seems fine, but since asSeenFrom itself uses [TypeOps.isLegalPrefix](https://github.com/lampepfl/dotty/blob/db170aaa01a28957575b72d9476e46f5671ce0ef/compiler/src/dotty/tools/dotc/core/TypeOps.scala#L126-L127) for this purpose, wouldn't it make sense to also use it here to ensure the checks stay aligned?